src/app/modules/trade-companion/component/trade-notification/trade-notification.component.ts

```diff
@@ -291,6 +298,26 @@ export class TradeNotificationComponent implements OnInit, OnDestroy, OnChanges
     this.dismiss()
   }
 
+  private highlight(): void {
+    if (this.notification.itemLocation != null) {
+      this.stashService.highlight(`pos:${this.notification.itemLocation.bounds.x},${this.notification.itemLocation.bounds.y}`).subscribe()
+    } else {
+      let highlightText = this.notification.item
+      if (typeof highlightText === 'object') {
```

Comment:
This if-statement should be redundant, the notification item is always of the type `CurrencyAmount` when the `itemLocation` is undefined.  
Thus you can always directly declare `let highlightText = (this.notification.item as CurrencyAmount).currency.nameType`